sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/LRUCache.java

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
```

Comment:
doesn't guava library that is shaded into our code have a LRUCache we can re-use?

Reply:
guava has a CompactLinkedHashMap which they say is more efficient than LinkedHashMap, and that could be used as LRUCache, but that class is not public